src/wallet/wallet2.cpp

```diff
@@ -1206,8 +1206,22 @@ void wallet2::refresh(uint64_t start_height, uint64_t & blocks_fetched, bool& re
   get_short_chain_history(short_chain_history);
   m_run.store(true, std::memory_order_relaxed);
   if (start_height > m_blockchain.size() || m_refresh_from_block_height > m_blockchain.size()) {
-    if (!start_height)
+    if (!start_height) {
+
+      std::string daemon_height_err = "";
+      uint64_t daemon_bc_height = this->get_daemon_blockchain_height(daemon_height_err);
+      if(daemon_height_err.size() > 0) {
+        throw std::runtime_error(daemon_height_err);
```

Comment:
That should throw another exception, so catch blocks match the actual error. But then it's probably due to get_daemon_blockchain_height not throwing the right one in the first place so I get where you're coming from :)